okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -659,13 +659,16 @@ private void fireTimer(Tenant tenant, ModuleDescriptor md, RoutingEntry re, Stri
       pl = new PermissionListV1(moduleTo, mdTo.getPermissionSets());
     } else if (permIntVer.equals("1.1")) {
       pl = new PermissionListV1(moduleTo, mdTo.getExpandedPermissionSets());
-    } else {
+    } else if (permIntVer.equals("2.0")) {
       if (mdFrom != null) {
         pl = new PermissionListV2(moduleTo, mdFrom.getId(), mdTo.getExpandedPermissionSets(),
             mdFrom.getExpandedPermissionSets());
       } else {
         pl = new PermissionListV2(moduleTo, mdTo.getExpandedPermissionSets());
       }
+    } else {
+      return Future.failedFuture(new OkapiError(ErrorType.ANY,
+          "Unknown version of _tenantPermissions interface in use " + permIntVer + "."));
     }
```

Comment:
I prefer you use ErrorType.USER here so that it returns 400, not 500. I think we should only return 500 when Okapi makes unexpected failures that are "not meant to happen".

Reply:
fair enough